scroll-gateway/src/ScrollProofService.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable prettier/prettier */
+import { EVMProofHelper, type IProofService } from '@ensdomains/evm-gateway';
+import { AbiCoder, concat, ethers, type AddressLike, } from 'ethers';
+
+export interface ScrollProvableBlock {
+    number: number
+}
+
+/**
+ * The proofService class can be used to calculate proofs for a given target and slot on the Scroll network.
+ * It's also capable of proofing long types such as mappings or string by using all included slots in the proof.
+ *
+ */
+export class ScrollProofService implements IProofService<ScrollProvableBlock> {
+    private readonly l2Provider: ethers.JsonRpcProvider;
+    private readonly helper: EVMProofHelper;
+
+    constructor(
+        l2Provider: ethers.JsonRpcProvider,
+    ) {
+        this.l2Provider = l2Provider;
+        this.helper = new EVMProofHelper(l2Provider);
+    }
+
+    async getStorageAt(block: ScrollProvableBlock, address: AddressLike, slot: bigint): Promise<string> {
+        return this.helper.getStorageAt(block.number, address, slot);
+    }
+
+
+    /**
+     * @dev Fetches a set of proofs for the requested state slots.
+     * @param block A `ProvableBlock` returned by `getProvableBlock`.
+     * @param address The address of the contract to fetch data from.
+     * @param slots An array of slots to fetch data for.
+     * @returns A proof of the given slots, encoded in a manner that this service's
+     *   corresponding decoding library will understand.
+     */
+    async getProofs(
+        block: ScrollProvableBlock,
+        address: AddressLike,
+        slots: bigint[]
+    ): Promise<string> {
+        const searchUrl = 'https://sepolia-api-re.scroll.io/api/search';
```

Comment:
This definitely needs to be a constructor argument, and ultimately supplied via the command line.